carbon-design-system/carbon
# chore(tile): refactor to functional component

REF #9712 

Refactored most of the Tile component to make functional components using `React.forwardRef` to ensure refs were being forwarded appropriately.

- [ ] Tile
- [ ] ClickableTile
- [ ] ~ExpandableTile~
- [ ] TileAboveTheFoldContent
- [ ] TileBelowTheFoldContent

#### Notes

- I outright gutted the `getDerivedStateFromProps` from `ClickableTile`. Everything seems still be working without it and the [guidance from React](https://reactjs.org/docs/hooks-faq.html#how-do-i-implement-getderivedstatefromprops) suggests that you rarely need it.


#### Testing / Reviewing

1. Pull down the PR locally.

2. Run `yarn start:v11` from `packages/react`

4. Ensure that the following stories for Tile are all working as expected: 
- [ ] `Default`
- [ ]  `ClickableTile`
- [ ]  `Multi Select`

Note: I have not refactored `ExpandableTile` in this PR.
